subscriptions/providers/apple_in_app/api.py

```diff
@@ -33,6 +36,8 @@ class Config:
     purchase_date: datetime.datetime = Field(alias='purchase_date_ms')
     # From documentation: The time a subscription expires or when it will renew.
     expires_date: datetime.datetime = Field(alias='expires_date_ms')
+    # Only available if it was cancelled or refunded.
+    cancellation_date: Optional[datetime.datetime] = Field(alias='cancellation_date_ms', default=None)
```

Comment:
But `cancellation_date_ms` sorta says we should expect it to be `int` or `float`, why `datetime.datetime`?

Reply:
All of these are provided as timestamps, but why use timestamps when we can use datetimes?

Reply:
But pydantic model describes external data structure, not what you want it to be. So when pydantic's `parse_obj` sees timestamp as input but expects datetime, it will raise validation error.

Reply:
https://pydantic-docs.helpmanual.io/usage/types/#datetime-types

```
datetime fields can be:
datetime, existing datetime object
int or float, assumed as Unix time, i.e. seconds (if >= -2e10 or <= 2e10) or milliseconds (if < -2e10or > 2e10) since 1 January 1970
str, following formats work:
YYYY-MM-DD[T]HH:MM[:SS[.ffffff]][Z or [±]HH[:]MM]]]
int or float as a string (assumed as Unix time)
```

Reply:
Cool, i didn't know this